src/primitives/plane.jl

```diff
@@ -0,0 +1,15 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Plane(p, n)
+
+A plane coincident with point `p`, with normal vector `n`.
+"""
+struct Plane{Dim,T} <: Primitive{Dim,T}
+  p::Point{Dim,T}
+  n::Vec{Dim,T}
+end
+
+paramdim(::Type{<:Plane}) = 2
```

Comment:
The parametric dimension of a plane is Dim-1 more generally.

Reply:
I'd taken the plane as a general 2-dimensional subspace in the same way that `paramdim(::Line{Dim}) == 1` for arbitrary `Dim`. Would this generalized treatment fit better under a new `Hyperplane` abstract type containing both `Line` and `Plane`?

Reply:
A `Line` is an actual 1D object defined by a direction. I don't see it as a plane. On the other hand, a Plane is defined here in terms of all points perpendicular to a direction, and this leads to a parametric dimension Dim-1.

Reply:
I would disagree. The orthogonal vector is like a dual of the plane in 3D. But inherently a plane is a combination of two vectors, and you need two parameters to navigate on it.

Reply:
My comment above is not consistent with the code I realized. I would suggest to define a plane with two vectors.

Reply:
So we have a lot of flexibility with these definitions, but we need to think in terms of the application. We can represent a plane in multiple ways and give it different semantics depending on the usage. I think we don't want to overcomplicate things. Should we open an issue to discus this Line/Plane situation in more detail?

Reply:
To add to the mix, a plane can also be seen as a sphere with infinite radius. In that case, it requires 4 points including one at infinity.

I agree that opening an issue on the matter would be helpful. Especially for such a basic primitive, we want to get it right and not suffer from making the wrong choices. I think a lot of stuff can be built on top of it.

Reply:
Following [this](https://github.com/JuliaGeometry/Meshes.jl/issues/90#issuecomment-804445499) discussion, I kept the original `paramdim(::Type{<:Plane}) = 2`. Is that okay, or do you still want to go with `paramdim(::Type{<:Plane{Dim,T}}) where {Dim} = Dim - 1`?